apps/cmp.c

```diff
@@ -1510,8 +1517,7 @@ static SSL_CTX *setup_ssl_ctx(OSSL_CMP_CTX *ctx, const char *host,
          * If we did this before checking our own TLS cert
          * the expected hostname would mislead the check.
          */
-        if (!truststore_set_host_etc(trust_store,
-                                     opt_tls_host != NULL ? opt_tls_host : host))
+        if (!truststore_set_host_etc(trust_store, host))
```

Comment:
Not sure I understand the purpose of this change. Can you explain it?

Reply:
`setup_ssl_ctx()` is called by `setup_client_ctx()` if TLS is used.
The change moves the conditional expression to the latter function
because the hostaddr is now needed also for setting up `info->sni_hostname` there.
This is for correct and consistent use of the host for hostname validation and SNI.